### PR TITLE
Add `PFCOUNT` command.

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
https://redis.io/commands/pfcount/

This PR depends on #166

### CODE REVIEW

File: java/client/src/main/java/glide/api/commands/HyperLogLogBaseCommands.java [hunk 1]
```diff
@@ -48,4 +48,26 @@ public interface HyperLogLogBaseCommands {
      * }</pre>
      */
     CompletableFuture<Long> pfadd(String key, String[] elements);
+
+    /**
+     * Gets the approximated cardinality (size) of the union of the HyperLogLogs passed, by internally
+     * merging into a temporary HyperLogLog.
```
Comment (SanHalacogluImproving) [inline java/client/src/main/java/glide/api/commands/HyperLogLogBaseCommands.java:54]:
```suggestion
     *Estimates the cardinality of data stored in a HyperLogLog structure for a single key or calculates the combined cardinality of multiple keys by merging their HyperLogLogs temporarily.
```

File: java/client/src/main/java/glide/api/commands/HyperLogLogBaseCommands.java [hunk 2]
```diff
@@ -48,4 +48,26 @@ public interface HyperLogLogBaseCommands {
...
+     *
+     * @see <a href="https://redis.io/commands/pfcount/">redis.io</a> for details.
+     * @param keys The data structure(s) to count cardinality.
```
Comment (SanHalacogluImproving) [inline java/client/src/main/java/glide/api/commands/HyperLogLogBaseCommands.java:57]: Identifiers for the HyperLogLog data structures to be analyzed.

File: java/client/src/main/java/glide/api/commands/HyperLogLogBaseCommands.java [hunk 3]
```diff
@@ -48,4 +48,26 @@ public interface HyperLogLogBaseCommands {
...
+     *     the variable does not exist.
+     * @example
+     *     <pre>{@code
+     * Long result = client.pfcount("hll_1", "hll_2").get();
+     * assert result == 42L; // Count of unique elements in multiple data structures
+     *
+     * result = client.pfcount("empty_hll").get();
+     * assert result == 0L;
+     *
+     * result = client.pfcount("not_existing_hll").get();
```
Comment (SanHalacogluImproving) [inline java/client/src/main/java/glide/api/commands/HyperLogLogBaseCommands.java:68]: You don't need to show every possible example in my opinion this can get out of hand for other commands.

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 4]
```diff
@@ -1270,6 +1271,21 @@ public T pfadd(String key, String[] elements) {
         return getThis();
     }
 
+    /**
+     * Gets the approximated cardinality (size) of the union of the HyperLogLogs passed, by internally
```
Comment (SanHalacogluImproving) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:1275]: Comments as above for documentation.

File: java/integTest/src/test/java/glide/TransactionTestUtilities.java [hunk 5]
```diff
@@ -92,8 +95,19 @@ public static BaseTransaction<?> transactionTest(BaseTransaction<?> baseTransact
...
-        // baseTransaction.pfadd(hllKey1, new String[0]);
+        baseTransaction
+            .pfadd(hllKey1, new String[0])
+            .pfadd(hllKey1, new String[] {"a", "b", "c"})
+            .pfadd(hllKey2, new String[] {"b", "c", "d"})
+            .pfadd(hllKey1, new String[0])
+            .pfadd(hllKey3, new String[0]);
+
+        baseTransaction
+            .pfcount(new String[] { "{key}:" + UUID.randomUUID()}) // TODO rename after #160
```
Comment (SanHalacogluImproving) [inline java/integTest/src/test/java/glide/TransactionTestUtilities.java:106]: We should have just one test for transactions we can test functionality in shared commands. (This is consistent with Py and Node and Java)

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 6]
```diff
@@ -1270,6 +1271,21 @@ public T pfadd(String key, String[] elements) {
...
+     * merging into a temporary HyperLogLog.
+     *
+     * @see <a href="https://redis.io/commands/pfcount/">redis.io</a> for details.
+     * @param keys The data structure(s) to count cardinality.
+     * @return Command Response - The approximated cardinality of given HyperLogLogs data structures
+     *     or <code>0</code> if the variable does not exist.
+     */
+    public T pfcount(String[] keys) {
```
Comment (SanHalacogluImproving) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:1283]: @nonull tag

**Comment by Yury-Fridlyand** [comment]
Round

**Review by jonathanl-bq — APPROVED** [review]
Some minor comments, but otherwise LGTM.

**Review by aaron-congo — APPROVED** [review]

File: CHANGELOG.md [hunk 7]
```diff
@@ -27,6 +27,7 @@
 * Core: Enabled Cluster Mode periodic checks by default ([#1089](https://github.com/aws/glide-for-redis/pull/1089))
 * Node: Added Rename command. ([#1124](https://github.com/aws/glide-for-redis/pull/1124))
 * Core: Added `PFADD` command. (TODO ADD HERE PR #)
+* Core: Added `PFCOUNT` command. (TODO ADD HERE PR #)
```
[the remark this answers is not in the document]
Comment (Yury-Fridlyand) [reply CHANGELOG.md:30]: Will add upstream PR # when I get it

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 8]
```diff
@@ -1029,4 +1029,27 @@ public void pfadd(BaseClient client) {
...
+        String key2 = "{test}-hll2-" + UUID.randomUUID();
+        String key3 = "{test}-hll3-" + UUID.randomUUID();
+        assertEquals(1, client.pfadd(key1, new String[] {"a", "b", "c"}).get());
+        assertEquals(1, client.pfadd(key2, new String[] {"b", "c", "d"}).get());
+        assertEquals(3, client.pfcount(new String[] {key1}).get());
+        assertEquals(3, client.pfcount(new String[] {key2}).get());
+        assertEquals(4, client.pfcount(new String[] {key1, key2}).get());
+        // empty HyperLogLog data set
+        assertEquals(1, client.pfadd(key3, new String[0]).get());
+        assertEquals(0, client.pfcount(new String[] {key3}).get());
```
Comment (acarbonetto) [inline java/integTest/src/test/java/glide/SharedCommandTests.java:1047]: add this to line 1044. 
```
assertEquals(4, client.pfcount(new String[] {key1, key2, key3}).get());
```

File: java/integTest/src/test/java/glide/TransactionTestUtilities.java [hunk 9]
```diff
@@ -92,8 +95,14 @@ public static BaseTransaction<?> transactionTest(BaseTransaction<?> baseTransact
 
         baseTransaction.configResetStat();
 
-        // TODO
-        // baseTransaction.pfadd(hllKey1, new String[0]);
+        baseTransaction
+                .pfadd(hllKey1, new String[0])
```
Comment (acarbonetto) [inline java/integTest/src/test/java/glide/TransactionTestUtilities.java:99]: we don't need to test the zero case.

File: java/integTest/src/test/java/glide/TransactionTestUtilities.java [hunk 10]
```diff
@@ -92,8 +95,14 @@ public static BaseTransaction<?> transactionTest(BaseTransaction<?> baseTransact
...
+                .pfadd(hllKey1, new String[] {"a", "b", "c"})
```
Comment (acarbonetto) [inline java/integTest/src/test/java/glide/TransactionTestUtilities.java:100]: just test for one HyperLogLog

Comment (Yury-Fridlyand) [reply java/integTest/src/test/java/glide/TransactionTestUtilities.java:100]: It is already reverted in #166

File: java/integTest/src/test/java/glide/TransactionTestUtilities.java (hunk 9 above)
Comment (Yury-Fridlyand) [reply java/integTest/src/test/java/glide/TransactionTestUtilities.java:99]: It is already reverted in https://github.com/Bit-Quill/glide-for-redis/pull/166

File: java/integTest/src/test/java/glide/SharedCommandTests.java (hunk 8 above)
Comment (Yury-Fridlyand) [reply java/integTest/src/test/java/glide/SharedCommandTests.java:1048]: Added

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 11]
```diff
@@ -1270,6 +1271,21 @@ public T pfadd(String key, String[] elements) {
         return getThis();
     }
 
+    /**
+     * Estimates the cardinality of the data stored in a HyperLogLog structure for a single key or
```
[the remark this answers is not in the document]
Comment (Yury-Fridlyand) [reply java/client/src/main/java/glide/api/models/BaseTransaction.java:1275]: Updated

File: java/client/src/main/java/glide/api/commands/HyperLogLogBaseCommands.java [hunk 12]
```diff
@@ -48,4 +48,20 @@ public interface HyperLogLogBaseCommands {
...
     CompletableFuture<Long> pfadd(String key, String[] elements);
+
+    /**
+     * Estimates the cardinality of the data stored in a HyperLogLog structure for a single key or
+     * calculates the combined cardinality of multiple keys by merging their HyperLogLogs temporarily.
+     *
+     * @see <a href="https://redis.io/commands/pfcount/">redis.io</a> for details.
+     * @param keys Identifiers of the HyperLogLog data structures to be analyzed.
+     * @return The approximated cardinality of given HyperLogLogs data structures or <code>0</code> if
+     *     the variable does not exist.
```
[the remark this answers is not in the document]
Comment (Yury-Fridlyand) [reply java/client/src/main/java/glide/api/commands/HyperLogLogBaseCommands.java]: Fixed

File: java/client/src/main/java/glide/api/commands/HyperLogLogBaseCommands.java [hunk 13]
```diff
@@ -48,4 +48,20 @@ public interface HyperLogLogBaseCommands {
...
      */
     CompletableFuture<Long> pfadd(String key, String[] elements);
+
+    /**
+     * Estimates the cardinality of the data stored in a HyperLogLog structure for a single key or
+     * calculates the combined cardinality of multiple keys by merging their HyperLogLogs temporarily.
+     *
+     * @see <a href="https://redis.io/commands/pfcount/">redis.io</a> for details.
+     * @param keys Identifiers of the HyperLogLog data structures to be analyzed.
+     * @return The approximated cardinality of given HyperLogLogs data structures or <code>0</code> if
```
[the remark this answers is not in the document]
Comment (Yury-Fridlyand) [reply java/client/src/main/java/glide/api/commands/HyperLogLogBaseCommands.java]: Fixed

File: java/client/src/main/java/glide/api/commands/HyperLogLogBaseCommands.java [hunk 14]
```diff
@@ -48,4 +48,20 @@ public interface HyperLogLogBaseCommands {
      * }</pre>
      */
     CompletableFuture<Long> pfadd(String key, String[] elements);
+
+    /**
+     * Estimates the cardinality of the data stored in a HyperLogLog structure for a single key or
+     * calculates the combined cardinality of multiple keys by merging their HyperLogLogs temporarily.
+     *
+     * @see <a href="https://redis.io/commands/pfcount/">redis.io</a> for details.
+     * @param keys Identifiers of the HyperLogLog data structures to be analyzed.
```
[the remark this answers is not in the document]
Comment (Yury-Fridlyand) [reply java/client/src/main/java/glide/api/commands/HyperLogLogBaseCommands.java]: Fixed

File: CHANGELOG.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (aaron-congo) [reply CHANGELOG.md:30]: Apparently we aren't supposed to update the CHANGELOG since Java isn't released yet, I asked about this on slack

[the remark this answers is not in the document]
Comment (Yury-Fridlyand) [reply CHANGELOG.md:30]: We need a changelog entry for core

**Review by SanHalacogluImproving — APPROVED** [review]
